src/test/java/harness/TestUser.java

```diff
@@ -62,27 +78,92 @@
    * @return global context object, populated with the user's credentials
    */
   public void login() throws IOException {
+    System.out.println("Logging in test user: " + email);
+
     // get domain-wide delegated credentials for this user. use the same scopes that are requested
     // of CLI users when they login.
-    System.out.println("Logging in test user: " + email);
-    GoogleCredentials delegatedUserCredential = getCredentials(User.USER_SCOPES);
+    GoogleCredentials googleCredentials = getCredentials(User.USER_SCOPES);
+    writeTerraOAuthCredentialFile(googleCredentials);
+
+    // We're not using pet SA key file (for security reasons), so auth is more complicated.
```

Comment:
// We're authenticating as an end user, not a pet SA?

In general I'm making these comments because I feel like we should describe what we're authenticating as, where as saying "not SA" seems only relevant if you knew how it previously worked?

Reply:
I think these comments are useful if there's a new tool, or if what's we're doing for bq/gsutil breaks. GCP only documents SA key way / tools only cleanly support SA key way. It let's person know figuring out auth for a new tool might be hard.